lib/services/elasticsearch.js

```diff
@@ -799,7 +807,15 @@ class ElasticSearch extends Service {
   getMapping(request) {
     const esRequest = initESRequest(request);
 
-    return this.esWrapper.getMapping(esRequest);
+    const includeKuzzleInfo = (() => {
```

Comment:
Yes but when you use this parameter in an url `?includeKuzzleInfo=false` you will never have a boolean value, only a string. This is why I have to do this complex boolean calculation.

By the way, we have the same misbehaviour with `?pretty=false`

Reply:
I don't like dealing with HTTP specifics outside the HTTP protocol code.

Actually, you mention the `pretty` option, and it is a very good example:

```
$ curl 'http://localhost:7512/users/_me?pretty=false'                                                     
{
  "requestId": "b705dc66-c6b9-41ba-a951-38785b40c856",
  "status": 200,
  "error": null,
  "controller": "auth",
  "action": "getCurrentUser",
  "collection": null,
  "index": null,
  "volatile": null,
  "result": {
    "_id": "-1",
    "_source": {
      "profileIds": [
        "anonymous"
      ],
      "name": "Anonymous"
    },
    "_meta": {},
    "strategies": []
  }
}
```

This is because in URLs we decided to deal with toggles (what booleans are) as.. toggles: if you put the option in the querystring, then it's set. If you want to unset the option, you remove it from the URL.

So, about `pretty`: Kuzzle ignores any value you put to it. As long as it's there, Kuzzle pretty prints.

So, if we go that way, your code can be simplified like this:

```js
const includeKuzzleInfo = Boolean(request.input.args.includeKuzzleInfo);
```

Also, my remark about your code being complicated is mainly about the anonymous function that you create and execute just after that, I don' get the reason behind that.

Reply:
After giving a long, hard think about it, I guess the only way to deal with this is to do this:

```js
let includeKuzzleInfo;
if (!request.context.connection.protocol === 'http' 
  && !_.isNil(request.input.args.includeKuzzleInfo) 
  && typeof request.input.args.includeKuzzleInfo !== 'boolean') {

  throw new BadRequestError(`Invalid includeKuzzleInfo value ${request.input.args.includeKuzzleInfo}: boolean expected`);
}

const includeKuzzleInfo = Boolean(request.input.args.includeKuzzleInfo);
```

Reply:
I had to do slightly differently because an empty string is considered as a `false` value.
`const includeKuzzleInfo = request.input.args.includeKuzzleInfo !== undefined ? true : false;`

I found strange to deal with this kind of protocol specific code inside the elasticsearch service. I think that the elasticsearch service should not be aware of our `Request` object. The controllers should extract the different parameters and then pass them to the service. 
`getMapping (index, collection, includeKuzzleInfo = false)` should be the method signature

Reply:
@Aschen > One last question before we merge this PR: I wonder if checking for HTTP specifics wouldn't be better made in the controllers invoking that method?

This is a last minute thought I just got. Tell me how you feel about it, so I can give my validation and finally get this much awaited feature a go ;-)

Reply:
Yes I agree that this will be better but still I think we should refactor the way we extract param from the request and do this in the controllers

Reply:
This may warrant a workshop, there is a lot to say about this subject (I do, at least ;-) ).